lib/march_hare/channel.rb

```diff
@@ -247,10 +254,11 @@ def recover_tx_mode
     def recover_exchanges
       @exchanges.values.each do |x|
         begin
+          logger.debug("channel: recover exchange #{x.name}")
           x.recover_from_network_failure
         rescue Exception => e
-          # TODO: logger
-          $stderr.puts "Caught exception when recovering exchange #{x.name}"
+          logger.info("Caught exception when recovering exchange #{x.name}")
+          logger.error(e)
```

Comment:
So I do 2 calls to the logger in all cases where I have a message + a exception object so that the logger implementation can decide itself how it wants to serialize the exception.

Reply:
Fair enough.